client/src/components/FileStatusIcon/FileStatusIcon.js

```diff
@@ -0,0 +1,112 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import i18n from 'i18n';
+import { UncontrolledTooltip } from 'reactstrap';
+
+/**
+ * File status icon (e.g. restricted access) with a tooltip
+ */
+class FileStatusIcon extends Component {
+  constructor(props) {
+    super(props);
+    this.cssClass = 'file-status-icon';
+  }
+
+  /**
+   * @param {boolean} hasRestrictedAccess
+   * @returns {Object}
+   */
+  buildUserFormAttrs(hasRestrictedAccess) {
+    const fontIconClass = !hasRestrictedAccess
+      ? 'font-icon-address-card-warning'
+      : 'font-icon-address-card';
+    const className = `icon ${this.cssClass}__icon ${fontIconClass}`;
+    const dataTitle = !hasRestrictedAccess
+      ? i18n._t(
+        'SilverStripe\\Admin\\FileStatusIcon.USERFORM_UPLOAD_UNRESTRICTED',
+        'Form submission, unrestricted access'
+      )
+      : i18n._t(
+        'SilverStripe\\Admin\\FileStatusIcon.USERFORM_UPLOAD_RESTRICTED',
+        'Form submission'
+      );
+    return { className, 'data-title': dataTitle };
+  }
+
+  /**
+   * @param {boolean} hasRestrictedAccess
+   * @returns {Object}
+   */
+  buildRestrictedFileAttrs(hasRestrictedAccess) {
+    const fontIconClass = hasRestrictedAccess
+      ? 'font-icon-user-lock'
+      : ''; // no icon for unrestricted
+    const className = `icon ${this.cssClass}__icon ${fontIconClass} `.trim();
+    const dataTitle = hasRestrictedAccess
+      ? i18n._t(
+        'SilverStripe\\Admin\\FileStatusIcon.ACCESS_RESTRICTED',
+        'Restricted access'
+      )
+      : i18n._t(
+        'SilverStripe\\Admin\\FileStatusIcon.ACCESS_UNRESTRICTED',
+        'Public access'
+      );
+    return { className, 'data-title': dataTitle };
+  }
+
+  /**
+   * @param {string} placement
+   * @param {string} id
+   * @param {string} title
+   * @returns {*}
+   */
+  renderTooltip(placement, id, title) {
+    return (
+      <UncontrolledTooltip placement={placement} target={id} delay={{ show: 300, hide: 0 }}>
+        {title}
+      </UncontrolledTooltip>
+    );
+  }
+
+  /**
+   * @returns {*}
+   */
+  render() {
+    const {
+      fileID, hasRestrictedAccess, isUserFormUpload, placement, className,
+      disableTooltip, includeBackground
+    } = this.props;
+    const backgroundClass = includeBackground ? 'file-status-icon--background' : '';
+    const classes = [`${this.cssClass}`, backgroundClass, className];
+    const attrs = isUserFormUpload
+      ? this.buildUserFormAttrs(hasRestrictedAccess)
+      : this.buildRestrictedFileAttrs(hasRestrictedAccess);
+    const idType = isUserFormUpload ? 'userform-upload' : 'restricted';
+    const id = `FileStatusIcon-${idType}-${fileID}`;
+    const tooltip = disableTooltip ? '' : this.renderTooltip(placement, id, attrs['data-title']);
+    return (
+      <div className={classes.filter(x => x).join(' ')}>
```

Comment:
We already use `classnames` in this repo (and pretty heavily in presentational components), you might find that more helpful than custom `filter` and `join` calls.

Reply:
Updated to use classnames